Document file system object content addressing (#10479)
NixOS/nix: doc/manual/src/language/advanced-attributes.md

    The `outputHashMode` attribute determines how the hash is computed.
    It must be one of the following two values:

    <!-- FIXME link to store object content-addressing not file system object content addressing once we have the page for that. -->

      - `"flat"`

        The output must be a non-executable regular file; if it isn’t, the build fails.
        The hash is
        [simply computed over the contents of that file](@docroot@/store/file-system-object/content-address.md#serial-flat)
        (so it’s equal to what Unix commands like `sha256sum` or `sha1sum` produce).

        This is the default.

      - `"recursive"` or `"nar"`

        The hash is computed over the
        [Nix Archive (NAR)](@docroot@/store/file-system-object/content-address.md#serial-nix-archive)
        dump of the output (i.e., the result of [`nix-store --dump`](@docroot@/command-ref/nix-store/dump.md)).

>         In this case, the output can be anything, including a directory tree.

Reviewer: ```suggestion
        In this case, the output will be any [file system object].
```

- not _can_, because that's like talking about the derivation output producing process, which I don't think we want to get into here(?)
- file system object.

Perhaps more accessible:

```suggestion
        In this case, the output will be a [file system object], which may include directories, and more.
```

Author: I think this was just me reformatting the existing sentence :) but how about?
```suggestion
        In this case, the output is allowed to be any [file system object], including directories and more.
```

Since these docs are derivation specific, I think it is good to reference the process a bit. Otherwise we could simply link the new page and delete this all text.

Reviewer: "will be" was my attempt to bridge that.

Author: Yeah something about "will" with or without the "any" just doesn't feel grammatical to me, regardless of those two perspectives. A derivation output's root file system object can only be a single type of file file system object.

Author: I used `allowed` to hopefully describe invariant regardless of the process -- nevermind where it is enforced, the derivation up-front is making claims that will be true from a downstream (i.e. in event of succeeded build's) perspective.
